src/containers/single-event/SingleEvent.tsx

```diff
@@ -0,0 +1,92 @@
+import { Spin } from 'antd';
+import React, { useEffect } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import EventDetails from '../../components/event-details/EventDetails';
+import { C4CState } from '../../store';
+import {
+  AsyncRequest,
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+  asyncRequestIsNotStarted,
+} from '../../utils/asyncRequest';
+import { getUpcomingEvents } from '../upcoming-events/ducks/thunks';
+import { EventProps, EventsReducerState } from '../upcoming-events/ducks/types';
+
+const ContentContainer = styled.div`
+  padding: 24px;
+  margin: auto;
+  max-width: 1200px;
+  width: 80%;
+`;
+
+const CenteredContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+interface UpcomingEventsProps {
+  readonly events: EventsReducerState['upcomingEvents'];
+}
+
+interface SingleEventParams {
+  id: string;
+}
+
+const SingleEvent: React.FC<UpcomingEventsProps> = ({ events }) => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (asyncRequestIsNotStarted(events)) {
+      dispatch(getUpcomingEvents());
+    }
+  }, []);
+
+  const id = Number(useParams<SingleEventParams>().id);
```

Comment:
Still think we should separate out the hook to make its usage clear

Reply:
I think its ok, its just being casted so I dont think theres enough complexity to warrant it as another line. Im open to discussion though.